### PR TITLE
Replace LastErrorCode and LastErrorString with exceptions

### LINKED ISSUE #137: Exception vs LastErrorCode / LastErrorString
As the title says, use of `Exception`s vs `LastErrorCode` and `LastErrorString`. Personally I favor exceptions because they don't break under concurrent use and they're the builtin (as in within the language) solution for this problem. A custom exception type can be created to also return the error code. There are some cases where exceptions are thrown on precondition failures while in other cases the error code and message are set, I think this should be implemented consistently.

### PR DESCRIPTION
**Pull request by mycroes**
Fixes #137.

### CODE REVIEW

File: S7.Net/PlcSynchronous.cs [hunk 1]
```diff
@@ -547,15 +503,9 @@ public void ReadMultipleVars(List<DataItem> dataItems)
```
Comment (rapha-dev) [inline S7.Net/PlcSynchronous.cs]: Some Exceptions are still generic, should be `PlcException` here.

**Comment by mycroes** [comment]
If you hover over a line there's be a blue box with a plus near the start of the line, click it to comment inline.

**Comment by mycroes** [comment]
Fixed the 3 issues you pointed out.

To be honest this is not exactly how I would normally do exception handling, but this is closest to the behavior with the `LastErrorCode` and `LastErrorString`.

**Comment by rapha-dev** [comment]
It might be better to commit this PR to a new `development` branch since it will break existing projects without having a proper exception handling. It was necessary to monitor `ErrorCode` to ensure everything went well, now it is gone.

**Comment by mycroes** [comment]
I agree, keep in mind though that the only difference the develop branch will make is that it's easier to release hotfixes for the current codebase. I'm not sure if I can change the default branch, but my proposal would be to change the default branch to develop and merge only releases to master (taking GitFlow as branching model).

**Comment by rapha-dev** [comment]
Agreed, GitFlow also sounds good.

**Comment by rapha-dev** [comment]
Closing this as it would merge to master, replaced by #161

**Pull request closed without merging**